skfda/preprocessing/feature_construction/_function_transformers.py

```diff
@@ -0,0 +1,196 @@
+        the occupation measure
+        >>> from skfda.representation import FDataGrid
+        >>> import numpy as np
+        >>> t = np.linspace(0, 10, 100)
+        >>> fd_grid = FDataGrid(
+        ...     data_matrix=[
+        ...         t,
+        ...         2 * t,
+        ...         np.sin(t),
+        ...     ],
+        ...     grid_points=t,
+        ... )
+
+        Finally we call to the occupation measure function with the
+        intervals that we want to consider. In our case (0.0, 1.0)
+        and (2.0, 3.0). We need also to specify the number of points
+        we want that the function takes into account to interpolate.
+        We are going to use 501 points.
+        >>> from skfda.preprocessing.feature_construction import (
+        ...     OccupationMeasureTransformer,
+        ... )
+        >>> occupation_measure = OccupationMeasureTransformer(
+        ...     intervals=[(0.0, 1.0), (2.0, 3.0)],
+        ...     n_points=501,
+        ... )
+
+        >>> np.around(occupation_measure.fit_transform(fd_grid), decimals=2)
+        array([[ 0.98,  0.5 ,  6.28],
+               [ 1.02,  0.52,  0.  ]])
+    """
+
+    def __init__(
+        self,
+        intervals: Sequence[Tuple[float, float]],
+        *,
+        n_points: Optional[int] = None,
+    ):
+        self.intervals = intervals
+        self.n_points = n_points
+
+    def transform(self, X: Union[FDataGrid, FDataBasis]) -> NDArrayFloat:
+        """
+        Transform the provided data using the occupation_measure function.
+
+        Args:
+            X: FDataGrid or FDataBasis with the samples that are going to be
+                transformed.
+
+        Returns:
+            Array of shape (n_intervals, n_samples) including the transformed
+            data.
+        """
+        return occupation_measure(X, self.intervals, n_points=self.n_points)
+
+
+class NumberUpCrossingsTransformer(BaseEstimator, TransformerMixin):
+    """
+    Transformer that works as an adapter for the number_up_crossings function.
+
+    Args:
+        levels: sequence of numbers including the levels
+            we want to consider for the crossings.
+    Example:
+    For this example we will use a well known function so the correct
+    functioning of this method can be checked.
+    We will create and use a DataFrame with a sample extracted from
+    the Bessel Function of first type and order 0.
+    First of all we import the Bessel Function and create the X axis
+    data grid. Then we create the FdataGrid.
+    >>> from skfda.preprocessing.feature_construction import (
+    ...     NumberUpCrossingsTransformer,
+    ... )
+    >>> from scipy.special import jv
+    >>> from skfda.representation import FDataGrid
+    >>> import numpy as np
+    >>> x_grid = np.linspace(0, 14, 14)
+    >>> fd_grid = FDataGrid(
+    ...     data_matrix=[jv([0], x_grid)],
+    ...     grid_points=x_grid,
+    ... )
+    >>> fd_grid.data_matrix
+    array([[[ 1.        ],
```

Comment:
This should also need to be a 2D matrix.

Reply:
The output of OccupationMeasureTransformer is indeed a 2D array. What you are pointing is the data_matrix of a Bessel Function on a FDataGrid format.

Reply:
Sure thing! Sorry for the mistake.